lib/Driver/ToolChains/Hcc.cpp

```diff
@@ -59,6 +61,11 @@ void HCCInstallationDetector::AddHCCIncludeArgs(const llvm::opt::ArgList &Driver
   if (IsValid) {
     CC1Args.push_back(DriverArgs.MakeArgString("-I" + IncPath + "/include"));
     CC1Args.push_back(DriverArgs.MakeArgString("-I" + IncPath + "/hcc/include"));
+
+    if (DriverArgs.hasFlag(options::OPT_famdgpu_function_calls,
+                           options::OPT_fno_amdgpu_function_calls,
+                           FunctionCallDefault))
+      CC1Args.push_back("-famdgpu-function-calls");
```

Comment:
It doesn't seem like this is the right place to put this logic here since this is unrelated to include path processing.  Should this go into `construct_amdgpu_target_cmdargs` instead?

Reply:
I'm not sure if that function has access to the CC1 Args. That function only has Command arguments for Assembler and Linker, but this is a Frontend Option. The second part --amdgpu-func-calls can be moved to HCC:CXXAMPLINK::ConstructLinkerJob, but -famgpu-function-calls might need to go into Linux.cpp instead

Reply:
I've tested an alternative where I move --amdgpu-func-calls to `construct_amdgpu_target_cmdargs` but I move the -famdgpu-function-calls to `addClangTargetOptions`. another possibility is rename `addHCCIncludeArgs` to something generic like `addHCCCC1Args`